components/twitter/sources/common-followers.js

```diff
@@ -60,20 +62,52 @@ module.exports = {
         ts,
       };
     },
-    async pullFollowers() {
+    async provisionFollowersCache() {
       const screenName = this.getScreenName();
-      const followerIdsGen = this.twitter.getAllFollowers(screenName);
+      const followerIdsGen = this.twitter.scanFollowerIds(screenName);
+      const maxFollowerListSize = Math.max(this.getFollowersCacheSize(), 0);
       const result = [];
       for await (const id of followerIdsGen) {
+        if (maxFollowerListSize !== 0 && maxFollowerListSize === result.length) {
+          break;
+        }
+
         result.push(id);
       }
+
+      this.setFollowersCache(result);
       return result;
     },
+    setFollowersCache(followers) {
+      const followersCacheSize = Math.max(this.getFollowersCacheSize(), 0);
+      const trimmedFollowers = followers.slice(0, followersCacheSize);
```

Comment:
Won't this trim all the followers in the unlimited case?

Reply:
Ah yes, nice catch! I need to fix this.